corehq/apps/app_manager/models.py

```diff
@@ -1573,22 +1573,30 @@ def custom_icon_form_and_text_by_language(self, lang):
             return custom_icon.form, custom_icon_text
         return None, None
 
-    def _set_media(self, media_attr, lang, media_path):
+    def _set_media(self, media_attr, lang, media_path, app=None):
         """
             Caller's responsibility to save doc.
             Currently only called from the view which saves after all Edits
         """
         assert media_attr in ('media_image', 'media_audio')
 
         media_dict = getattr(self, media_attr) or {}
+        old_value = media_dict.get(lang)
+        # remove the entry from app multimedia mappings if media is being removed now
+        # This does not remove the multimedia but just it's reference in mapping
+        # Added it here to ensure it's always set instead of getting it only when needed
+        # ToDo: remove app argument and where ever its passed when called
```

Comment:
Could this todo be part of this PR?

Reply:
Rather, could addressing this todo be part of this PR?

Reply:
Sorry if it was not clear, yes. Would do that if the approach i took was approved so that i dont need to add it back in case it wasnt.

Reply:
hooray